text/0000-fabric-gateway.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: default
+title: Fabric Gateway
+nav_order: 3
+---
+
+- Feature Name: Fabric Gateway
+- Start Date: 2020-09-08
+- RFC PR: (leave this empty)
+- Fabric Component: fabric-gateway
+- Fabric Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The Fabric Gateway is a new component that will run either as a standalone process, or embedded with the peer, and will implement as much of the high-level 'gateway' programming model as possible.  This will remove much of the transaction submission logic from the client application which will simplify the maintenance of the SDKs.  It will also simplify the administrative overhead of running a Fabric network since the client application will only need to connect to a gateway rather than multiple peers across potentially multiple organisations.
```

Comment:
> It will also simplify the administrative overhead of running a Fabric network since the client application will only need to connect to a gateway rather than multiple peers across potentially multiple organisations.

I'm not sure how this improves administration of the network:

- You are now adding more nodes to the network, and of a new type, so you need new scripts and configuration.
- The gateways are going to be discovery clients, so hypothetically instead of `n` clients doing discovery queries, you have `k>n` (`k` gateways) doing it, but, this doesn't really improve administration, because if `k` gateways can do it, so do all `n` clients, and if `n` clients cannot do it, due to some environmental issue, then so can't `k` gateways. 
- It only adds another layer which can potentially fail, and now SREs need to know to diagnose the gateways as well.

Reply:
I agree that this introduces additional server-side elements to be managed. Hopefully we can minimise the management effort required.

I don't expect there to be anywhere near the number of Gateways as there are clients. Each Gateway instance will serve many clients. I don't agree with your `k>n` assertion.

Today you have to punch holes in your firewall to allow all clients (including ones outside of your organization) to access all of your nodes. And all of those network clients potentially establish multiple connections to all of those nodes, which presents administrative and scaling challenges. The proposed Gateway model means you only need to allow ingress for your own organization clients to your own Gateway node(s) at a single external endpoint.

Reply:
I meant `<`, that `>` was a typo :) 

I understand the firewall issue.

Reply:
If we only aim to achieve client -> gateway -> fabric peer/orderer, fabric providers have already well prepared with firewall solution, satisfying `you only need to allow ingress for your own organization clients to your own Gateway node(s) `
Provider nowadays has gateway node design there already.